models/system/setting.go

```diff
@@ -268,13 +275,13 @@ func Init() error {
 	if setting_module.OfflineMode {
 		disableGravatar = true
 		enableFederatedAvatar = false
-		if !GetSettingBool(KeyPictureDisableGravatar) {
-			if err := SetSettingNoVersion(KeyPictureDisableGravatar, "true"); err != nil {
+		if !GetSettingBool(db.DefaultContext, KeyPictureDisableGravatar) {
```

Comment:
Isn't it harmful to use `db.DefaultContext` now because this uses a context cache without request lifetime? `db.DefaultContext` is valid for the whole application lifetime. I try to replace #22491 but there `db.DefaultContext` would be used sometimes.

Reply:
Here we use `db.DefaultContext` then there is no request level cache will be triggered. The context cache will be ignored here.